types/vhost/directory.pp

```diff
@@ -0,0 +1,68 @@
+type Apache::Vhost::Directory = Struct[{
+    'allow'                                               => Optional[String],
+    'deny'                                                => Optional[String],
+    'order'                                               => Optional[String],
+    'satify'                                              => Optional[String],
+    'path'                                                => Optional[String],
+    'provider'                                            => Optional[String],
+    'headers'                                             => Optional[Variant[String[1], Array[String[1]]]],
+    'geoip_enable'                                        => Optional[Apache::Bool],
+    'options'                                             => Optional[Variant[String[1], Array[String[1]]]],
+    'index_options'                                       => Optional[Variant[String[1], Array[String[1]]]],
+    'index_order_default'                                 => Optional[Variant[String[1], Array[String[1]]]],
+    'index_stylesheet'                                    => Optional[String],
+    'allow_override'                                      => Optional[Variant[String[1], Array[String[1]]]],
+    'satify'                                              => Optional[String],
+    'limit'                                               => Any, # TODO: complex type
```

Comment:
I'm wondering if I should take this approach with all keys: just start with listing them all with the Any type. Tighten validation from there.